discord/ui/section.py

```diff
@@ -0,0 +1,284 @@
+    """Represents a UI section. Sections must have 1-3 (inclusive) items and an accessory set.
+
+    .. versionadded:: 2.7
+
+    Parameters
+    ----------
+    *items: :class:`Item`
+        The initial items contained in this section, up to 3.
+        Currently only supports :class:`~discord.ui.TextDisplay`.
+    accessory: Optional[:class:`Item`]
+        The section's accessory. This is displayed in the top right of the section.
+        Currently only supports :class:`~discord.ui.Button` and :class:`~discord.ui.Thumbnail`.
+    """
+
+    __section_accessory_item__: ClassVar[ItemCallbackType] = None
+
+    def __init_subclass__(cls) -> None:
+        accessory: ItemCallbackType = None
+        for base in reversed(cls.__mro__):
+            for member in base.__dict__.values():
+                if hasattr(member, "__discord_ui_model_type__"):
+                    accessory = member
+
+        cls.__section_accessory_item__ = accessory
+
+    def __init__(self, *items: Item, accessory: Item = None, id: int | None = None):
+        super().__init__()
+
+        self.items: list[Item] = []
+        self.accessory: Item | None = None
+
+        self._underlying = SectionComponent._raw_construct(
+            type=ComponentType.section,
+            id=id,
+            components=[],
+            accessory=None,
+        )
+        if func := self.__section_accessory_item__:
+            item: Item = func.__discord_ui_model_type__(
+                **func.__discord_ui_model_kwargs__
+            )
+            item.callback = partial(func, self, item)
+            self.set_accessory(item)
+            setattr(self, func.__name__, item)
+        elif accessory:
+            self.set_accessory(accessory)
+        for i in items:
+            self.add_item(i)
+
+    def _add_component_from_item(self, item: Item):
+        self._underlying.components.append(item._underlying)
+
+    def _set_components(self, items: list[Item]):
+        self._underlying.components.clear()
+        for item in items:
+            self._add_component_from_item(item)
+
+    def add_item(self, item: Item) -> None:
+        """Adds an item to the section.
+
+        Parameters
+        ----------
+        item: :class:`Item`
+            The item to add to the section.
+
+        Raises
+        ------
+        TypeError
+            An :class:`Item` was not passed.
+        ValueError
+            Maximum number of items has been exceeded (10).
+        """
+
+        if len(self.items) >= 3:
+            raise ValueError("maximum number of children exceeded")
+
+        if not isinstance(item, Item):
+            raise TypeError(f"expected Item not {item.__class__!r}")
+
+        self.items.append(item)
+        self._add_component_from_item(item)
+        return self
+
+    def remove_item(self, item: Item | int) -> None:
+        """Removes an item from the section. If an int or str is passed, it will remove by Item :attr:`id` or ``custom_id`` respectively.

```

Comment:
Removed attr reference because I think it would link to `Section.id`
```suggestion
        """Removes an item from the section. If an :class:`int` or :class:`str` is passed,
        the item will be removed by Item ``id`` or ``custom_id`` respectively.
```